File: Python/List-2/sum13.py
# Return the sum of the numbers in the array, returning 0 for an empty array.
# Except the number 13 is very unlucky, so it does not count and numbers that
# come immediately after a 13 also do not count.

def sum13(nums):
  # Initialize the sum that will be returned
  sum = 0
  
  # Go over the array, exiting the loop once we find a 13
  i = 0
  while i < len(nums):
    if nums[i] == 13:
      i += 2
    else:
      sum += nums[i]
      i += 1
  
  # Return the sum
  return sum

File: Java/Array-2/sum13.java
/*
 * Return the sum of the numbers in the array, returning 0 for an empty array.
 * Except the number 13 is very unlucky, so it does not count and numbers that
 * come immediately after a 13 also do not count.
 */

public int sum13(int[] nums) {
  // Initialize the sum
  int sum = 0;
  
  // Go over the array, adding to the sum any elements that are not 13 or
  // are immediately after a 13
  for(int i = 0; i < nums.length; i++){
    if(nums[i] == 13){
      i++;
    }else{
      sum += nums[i];
    }
  }
  
  // Return the sum
  return sum;
}
